### PR TITLE
Replace pul_uv_rails

### PR DESCRIPTION
**Pull request by geekscruff**
See #3232

# Approach

- expect universalviewer at {app}/uv/universalviewer (public/uv/universalviewer); install there for CI with yarn package
- use iframe in view
- supply default uv.html and uv_config.json
- add documentation for how to install /configure / override

By using an <iframe>, the universalviewer javascript is self-contained, and need not impact on the main application, allowing for the use of different versions, for example of jquery. It is also easier to swap out, for example, if you have a universal viewer server, you can skip installing universalviewer and customize the `uv.html` to direct to the new universal viewer location. Examples provided by universalviewer use this <iframe> approach. 

# Suggested documentation:

## Universal Viewer

In order to display the Universal Viewer, Hyrax requires the Unviersal Viewer javascript library. Hyrax expects to find the library at `{application_url}/uv/universalviewer` (customarily in public/uv/universalviewer).

### Installing with YARN

The `package.json` file located in the application root directory is configured to install universalviewer version 3 into public/uv/universalviewer. The following command will install universalviewer:

```
yarn run install-uv
```

### Using a different Universal Viewer location

Hyrax uses an <iframe> to display Universal Viewer. The frame displays the default html page located at `public/uv.html`. 

There are two ways to use a different location for Universal Viewer:

1) Change the specified paths in `public/uv.html` file.
2) Override the `#universal_viewer_url_base` method in `app/helpers/hyrax/iiif_helper.rb` to point to a different viewer html location

### Universal Viewer Config

It is easy to configure Universal Viewer, for example to default the table of contents to be closed or to remove the download options. Hyrax provides a default (empty) configuration file at `public\uv_config.json`. Customise this file to change the Universal Viewer behavior. 

To use a different file altogether, override the `#universal_viewer_config_url` method in `app/helpers/hyrax/iiif_helper.rb`.

### CODE REVIEW

**Comment by geekscruff** [comment]
@no-reply failing with `.internal_test_app: No such file or directory` ... is there some circeci magic I need to do?

**Comment by cjcolvar** [comment]
@geekscruff If you rebase from master you should get an updated circleci config which might work better.

**Comment by jrochkind** [comment]
For the historical record, what made the iframe necessary, rather than an ordinary div on page as before?

File: spec/test_app_templates/lib/generators/test_app_generator.rb [hunk 1]
```diff
@@ -128,4 +128,9 @@ def configure_action_cable_to_use_redis
               "development:\n  adapter: async",
               "development:\n  adapter: redis\n  url: redis://localhost:6379"
   end
+  
+  def install_universal_viewer
+    exec('yarn run install-uv')
```
Comment (dlpierce) [inline spec/test_app_templates/lib/generators/test_app_generator.rb]:
```suggestion
    rake 'yarn:install'
```

**Pull request merged**